client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/email-validation-banner.tsx

```diff
@@ -0,0 +1,101 @@
+import { useTranslate } from 'i18n-calypso';
+import { useDispatch } from 'react-redux';
+import { useSendEmailVerification } from 'calypso/landing/stepper/hooks/use-send-email-verification';
+import { successNotice, errorNotice } from 'calypso/state/notices/actions';
+interface EmailValidationBannerProps {
+	email: string;
+	closeBanner: () => void;
+}
+
+const CheckmarkIcon = () => (
+	<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
+		<g clipPath="url(#clip0_1228_17585)">
+			<path
+				d="M18.3333 9.23355V10.0002C18.3323 11.7972 17.7504 13.5458 16.6744 14.9851C15.5984 16.4244 14.086 17.4773 12.3628 17.9868C10.6395 18.4963 8.79768 18.4351 7.11202 17.8124C5.42636 17.1896 3.98717 16.0386 3.00909 14.5311C2.03101 13.0236 1.56645 11.2403 1.68469 9.44714C1.80293 7.65402 2.49763 5.94715 3.66519 4.58111C4.83275 3.21506 6.41061 2.26303 8.16345 1.867C9.91629 1.47097 11.7502 1.65216 13.3916 2.38355"
+				stroke="#00A32A"
+				strokeWidth="2"
+				strokeLinecap="round"
+				strokeLinejoin="round"
+			/>
+			<path
+				d="M18.3333 3.33398L10 11.6757L7.5 9.17565"
+				stroke="#00A32A"
+				strokeWidth="2"
+				strokeLinecap="round"
+				strokeLinejoin="round"
+			/>
+		</g>
+		<defs>
+			<clipPath id="clip0_1228_17585">
+				<rect width="20" height="20" fill="white" />
+			</clipPath>
+		</defs>
+	</svg>
+);
+
+const CloseIcon = () => (
+	<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
+		<path
+			d="M3.60296 3.33398L12.6666 12.6673M3.33325 12.6673L12.3969 3.33398"
+			stroke="#A7AAAD"
+			strokeWidth="1.5"
+		/>
+	</svg>
+);
+
+const EmailValidationBanner = ( { email, closeBanner }: EmailValidationBannerProps ) => {
+	const resendEmailNotice = 'resend-verification-email';
+	const dispatch = useDispatch();
+	const resendEmail = useSendEmailVerification();
+	const translate = useTranslate();
+
+	const handleResend = async () => {
```

Comment:
@Addison-Stavlo I only need this function from the existing email nonce component (the rest is the specific markup). Do you think it makes sense to make that a reusable hook `useResendEmailVerification` and use it in both places? If so, where would be a good place to put it?